Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml

```diff
@@ -76,6 +76,56 @@
         reagents:
         - ReagentId: Water
           Quantity: 500
+  - type: ItemSlots
+    slots:
+      cup_slot_0:
+        name: Cup
+        startingItem: DrinkWaterCup
+        ejectOnInteract: true
+        insertOnInteract: false
+        whitelist:
+          tags:
+          - PaperCup
+      cup_slot_1:
+        name: Cup
+        startingItem: DrinkWaterCup
+        ejectOnInteract: true
+        insertOnInteract: false
+        whitelist:
+          tags:
+          - PaperCup
+      cup_slot_2:
+        name: Cup
+        startingItem: DrinkWaterCup
+        ejectOnInteract: true
+        insertOnInteract: false
+        whitelist:
+          tags:
+          - PaperCup
+      cup_slot_3:
+        name: Cup
+        startingItem: DrinkWaterCup
+        ejectOnInteract: true
+        insertOnInteract: false
+        whitelist:
+          tags:
+          - PaperCup
+      cup_slot_4:
+        name: Cup
+        startingItem: DrinkWaterCup
+        ejectOnInteract: true
+        insertOnInteract: false
+        whitelist:
+          tags:
+          - PaperCup
+      cup_slot_5:
+        name: Cup
+        startingItem: DrinkWaterCup
+        ejectOnInteract: true
+        insertOnInteract: false
+        whitelist:
+          tags:
+          - PaperCup
```

Comment:
Please don't. Make it a vending machine or something at least.

Reply:
Yea I getcha. I'm hoping to avoid a separate UI and have a single click to get it in your hand for fill up.
I'll look for another way to do this that doesn't involve a paragraph in the yml.

Reply:
I've added a new component to handle the desired behavior. Especially since Peptide said he would have use for it, there seemed to be value there.